wgsl/index.bs

```diff
@@ -454,6 +454,168 @@ whether failure to meet a particular requirement
 results in a shader-creation, pipeline-creation, or dynamic error.
 
 The WebGPU specification describes the consequences of each kind of error.
+A WGSL program with a [=shader-creation error=] or [=pipeline-creation error=] error will not be incorporated
+into a [=pipeline=] and hence will not be executed.
+
+## Diagnostics ## {#diagnostics}
+
+An implementation can generate [=diagnostics=] during [=shader module creation=].
+A <dfn noexport>diagnostic</dfn> is a message produced by the implementation for the benefit of the application author.
+
+We say a diagnostic is <dfn noexport>triggered</dfn> when a particular condition is met, known as the [=diagnostic/triggering rule=].
```

Comment:
I'm probably being super-pedantic (sorry, feel free to ignore), but this sort of implies that we only produce diagnostics for things that have named triggering rules. I assume that the shader-creation errors that all of the other validation rules in spec produce are also considered diagnostics, its just that their triggering rules are unnamed (and therefore uncontrollable)?

Reply:
You're right, we should be clear about that.  
Other errors should generate diagnostics, but they are not controllable via diagnostic filters.

Reply:
Should be addressed now.  PTAL.